code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -19,36 +8,91 @@
 	color = "#404030" // rgb: 64, 64, 48
 	nutriment_factor = 0
-	var/boozepwr = 10 //lower numbers mean the booze will have an effect faster.
+	var/boozepwr = 65 //Higher numbers equal higher hardness, higher hardness equals more intense alcohol poisoning
+
+/*
+Boozepwr Chart
+Note that all higher effects of alcohol poisoning will inherit effects for smaller amounts (i.e. light poisoning inherts from slight poisoning)
+In addition, severe effects won't always trigger unless the drink is poisonously strong
+All effects don't start immediately, but rather get worse over time; the rate is affected by the imbiber's alcohol tolerance
+
+0: Non-alcoholic
+1-10: Barely classifiable as alcohol - occassional slurring
+11-20: Slight alcohol content - slurring
+21-30: Below average - imbiber begins to look slightly drunk
+31-40: Just below average - no unique effects
+41-50: Average - mild disorientation, imbiber begins to look drunk
+51-60: Just above average - disorientation, vomiting, imbiber begins to look heavily drunk
+61-70: Above average - small chance of blurry vision, imbiber begins to look smashed
+71-80: High alcohol content - blurry vision, imbiber completely shitfaced
+81-90: Extremely high alcohol content - heavy toxin damage, passing out
+91-100: Dangerously toxic - extreme toxin damage
+*/
 
 /datum/reagent/consumable/ethanol/on_mob_life(mob/living/M)
-	M.jitteriness = max(M.jitteriness-5,0)
-	if(current_cycle >= boozepwr*0.5)
-		var/drunk_value = sqrt(volume*1000/boozepwr)
-		if(volume >= boozepwr*0.2)
-			if(M.slurring < drunk_value)
-				M.slurring += 4
-			M.Dizzy(drunk_value)
-		if(volume >= boozepwr*0.8)
-			if(M.confused < drunk_value)
-				M.confused += 3
-		if(volume >= boozepwr*3.8)
-			M.adjustToxLoss(1, 0)
-			. = 1
+	if(M.alcohol_tolerance == COMPLETE_TOLERANCE)
+		return ..() || .
+	var/alcohol_effect = (boozepwr * (current_cycle / 50)) / M.alcohol_tolerance
+	if(alcohol_effect >= 1)
+		if(prob(25))
+			M.slurring += 2
+		M.jitteriness = max(M.jitteriness - 3, 0)
+
+	if(alcohol_effect >= 11)
+		M.slurring += 1.2 //Constant but only a very slow increase
+
+	if(alcohol_effect >= 21)
+		M.drunk_state = LIGHTLY_DRUNK
+
+	if(alcohol_effect >= 41)
+		if(prob(25))
+			M.confused += 2
+		M.Dizzy(10)
+		M.drunk_state = DRUNK
+
+	if(alcohol_effect >= 51)
+		if(prob(15) && iscarbon(M))
+			var/mob/living/carbon/C = M
+			C.vomit(boozepwr / 5)
+		M.Dizzy(25)
+		M.confused += 1.2
+		M.drunk_state = HEAVILY_DRUNK
+
+	if(alcohol_effect >= 61)
+		if(prob(50))
+			M.adjust_blurriness(5)
+		M.drunk_state = SMASHED
+
+	if(alcohol_effect >= 71)
+		M.adjust_blurriness(1.2)
+		M.drunk_state = SHITFACED
+
+	if(alcohol_effect >= 81)
+		M.adjustToxLoss(boozepwr / 25)
+		if(prob(5))
+			M << "<span class='warning'>haha yeah</span>"
+			M.sleeping += 15
+
+	if(alcohol_effect >= 91)
+		M.adjustToxLoss(boozepwr / 20)
+		M.drunk_state = BARELY_ALIVE
+
+	if(alcohol_effect >= 101)
+		M.adjustToxLoss(boozepwr) //Let's be honest you shouldn't be alive by now
```

Comment:
Pretty quickly it does yeah, but the effect needs time to stack on both drinks so it wouldn't unless there was maybe 15-20 units of each